### PR TITLE
add librime

### PR DESCRIPTION
**Pull request by ksqsf**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -7,3 +7,6 @@
 [submodule "libthai"]
 	path = libthai
 	url = https://github.com/tlwg/libthai
+[submodule "librime"]
+	path = librime
+	url = git@github.com:rime/librime.git
```
Comment (eagleoflqj) [inline .gitmodules]: https

File: common.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ f5m_configure() {
   PKG_CONFIG_PATH=$HOMEBREW_PREFIX/lib/pkgconfig cmake -B build -G Ninja \
     -DCMAKE_INSTALL_PREFIX=$HOMEBREW_PREFIX \
     -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_FIND_ROOT_PATH="/Library/Input Methods/Fcitx5.app/Contents;$HOMEBREW_PREFIX" \
+    -DCMAKE_FIND_ROOT_PATH="/Library/Input Methods/Fcitx5.app/Contents;$HOMEBREW_PREFIX;$CMAKE_FIND_ROOT_PATH" \
```
Comment (eagleoflqj) [inline common.sh]: I don’t think it’s needed. There is no such `CMAKE_FIND_ROOT_PATH`.

File: scripts/librime.sh
Comment (eagleoflqj) [inline scripts/librime.sh:1]: lua and octagram plugins needed

**Review by eagleoflqj — CHANGES_REQUESTED** [review]

File: common.sh (hunk 2 above)
Comment (ksqsf) [reply common.sh]: It is used as an environment variable by librime.sh to pass marisa's build root to librime.

I will change it to a better name later.

Comment (ksqsf) [reply common.sh]: removed in favor of `CMAKE_PREFIX_PATH`

File: scripts/librime.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+rime_plugins=(hchunhui/librime-lua lotem/librime-octagram)
+for plugin in ${rime_plugins[@]}; do
+    bash install-plugins.sh $plugin
+done
+
+# Work around an upstream mistake. (hchunhui/librime-lua#300)
+(cd plugins/lua; sed -i '' 's/LUA_LIBRARIES/LUA_LDFLAGS/' CMakeLists.txt)
+
+# HACK: Force generation of rime.pc
+sed -i '' 's/MINGW/APPLE/' CMakeLists.txt
```
Comment (eagleoflqj) [inline scripts/librime.sh]: I'm going to adjust it at upstream directly since I have write access

**Review by eagleoflqj — APPROVED** [review]

**Pull request merged**